go-apps/meep-rnis/server/rnis.go

```diff
@@ -2746,9 +2798,128 @@ func populateL2Meas(key string, jsonInfo string, l2MeasData interface{}) error {
 		cellIndex = len(data.l2Meas.CellInfo) - 1
 	}
 
+	jsonPoaData, _ := rc[RNIS_DB_CONNECTOR_INDEX].JSONGetEntry(baseKey+"POA:"+ueData.ParentPoaName, ".")
+
+	latency := int32(0)
+	poaPacketLoss := int32(0)
+	if jsonPoaData != "" {
+		poaDataObj := convertJsonToPoaInfo(jsonPoaData)
+		if poaDataObj != nil {
+			latency = poaDataObj.Latency
+			ploss := poaDataObj.PacketLoss
+			//return between 10^-4 t 10^-6
+			ploss = ploss * 1000000 //10^-6
+			if ploss > 100 {
+				poaPacketLoss = 100
+			} else {
+				poaPacketLoss = int32(ploss)
+			}
+		}
+	}
+
+	ueStats := SumAppInfo{data.queryAppInsId, 0, 0, 0, 0}
+
+	//loop through each APP to get throuput
+	for _, appName := range ueData.AppNames {
+
+		//we calculate stats for the queried app only or for all if none provided
+		if appName != data.queryAppInsId && data.queryAppInsId != "" {
+			continue
+		}
+		keyName := msBaseKey + "*" + appName
+		appStats := SumAppInfo{appName, 0, 0, 0, 0}
+
+		err = rc[DATA_DB_CONNECTOR_INDEX].ForEachEntry(keyName, calculateSum, &appStats)
```

Comment:
Replace with a call to meep-metrics-store: `GetCachedNetworkMetric("*", appName)`

Reply:
fixed